test/replication-luatest/gh_9918_synchro_queue_additional_info_test.lua

```diff
@@ -0,0 +1,102 @@
+local fiber = require('fiber')
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+    local box_cfg = {
+        replication = {
+            server.build_listen_uri('server1', cg.replica_set.id),
+            server.build_listen_uri('server2', cg.replica_set.id),
+        },
+        replication_timeout = 0.1,
+    }
+    cg.leader = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = box_cfg,
+    }
+    cg.replica = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = box_cfg,
+    }
+    cg.replica_set:start()
+    cg.leader:exec(function()
+        box.ctl.promote()
+        local s = box.schema.space.create('s', {is_sync = true})
+        s:create_index('pk')
+    end)
+    cg.replica_set:wait_for_fullmesh()
+    cg.leader:wait_for_downstream_to(cg.replica)
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that new `age` field of `box.info.synchro.queue` works correctly.
+g.test_age_field = function(cg)
+    -- The synchronous queue is originally empty.
+    cg.leader:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+    cg.replica:exec(function()
+        t.assert_equals(box.info.synchro.queue.age, 0)
+    end)
+
+    -- Ensure that the synchronous transactions stay in the queue for a while.
+    cg.leader:update_box_cfg{replication_synchro_quorum = 3,
+                             replication_synchro_timeout = 60}
+    -- Add the first entry to the synchronous queue.
+    local fid1, oldest_entry_ts = cg.leader:exec(function()
+        local fiber = require('fiber')
+
+        local f = fiber.new(function() box.space.s:replace{0} end)
+        f:set_joinable(true)
+        return f:id(), fiber.clock()
+    end)
+    local wait_time = 1
+    fiber.sleep(wait_time)
+
+    cg.leader:exec(function(wait_time)
+        t.assert_almost_equals(box.info.synchro.queue.age, wait_time, 0.15)
```

Comment:
I'm afraid we cannot test the time like that. The test is already very flaky and it depends on hardware speed a lot. I propose to sleep for 0.6 seconds and then check, that wait time is > 0.5. After that we may add a new entry to limbo, sleep for 0.6 seconds more and check, that age is more than 1.

Reply:
Testing that the age > the wait time seems to also be unreliable, so I just test it is > 0 and < 120.